servicelayer/process.py

```diff
@@ -22,13 +23,14 @@ class ServiceQueue(object):
     PRIO_LOW = 1
     PRIORITIES = [PRIO_HIGH, PRIO_MEDIUM, PRIO_LOW]
 
-    def __init__(self, conn, operation, dataset, priority=PRIO_MEDIUM):
+    def __init__(self, conn, operation, job_id, dataset, priority=PRIO_MEDIUM):
         self.conn = conn
         self.operation = operation
+        self.job_id = job_id
```

Comment:
Should we have a helper function for making a good job id (basically just uuid4?). Could be `Job.random_id()`.